geo/src/algorithm/relate/geomgraph/intersection_matrix.rs

```diff
@@ -268,6 +283,226 @@ impl IntersectionMatrix {
+        && self.0[CoordPos::Outside][CoordPos::Inside] == Dimensions::Empty
+        && self.0[CoordPos::Outside][CoordPos::OnBoundary] == Dimensions::Empty
+    }
+
+    /// Returns `true` if `a` touches `b`: they have at least one point in common, but their
+    /// interiors do not intersect.
+    ///
+    /// # Notes
+    /// - Matches any of `[FT*******], [F**T*****], [F***T****]`
+    /// - This predicate is **symmetric**
+    #[allow(clippy::nonminimal_bool)]
+    pub fn is_touches(&self) -> bool {
+        // [FT*******]
+        self.0[CoordPos::Inside][CoordPos::Inside] == Dimensions::Empty
+        && self.0[CoordPos::Inside][CoordPos::OnBoundary] != Dimensions::Empty ||
+        // [F**T*****]
+        self.0[CoordPos::Inside][CoordPos::Inside] == Dimensions::Empty
+        && self.0[CoordPos::OnBoundary][CoordPos::Inside] != Dimensions::Empty ||
+        // [F***T****]
+        self.0[CoordPos::Inside][CoordPos::Inside] == Dimensions::Empty
+        && self.0[CoordPos::OnBoundary][CoordPos::OnBoundary] != Dimensions::Empty
+    }
+
+    /// Compares two geometry objects and returns `true` if their intersection "spatially crosses";
+    /// that is, the geometries have some, but not all interior points in common
+    ///
+    /// ```
+    /// use geo_types::{LineString, line_string, polygon};
+    /// use geo::relate::Relate;
+    ///
+    /// let line_string: LineString = line_string![(x: 85.0, y: 194.0), (x: 162.0, y: 135.0)];
+    /// let poly = polygon![
+    ///     (x: 125., y: 179.),
+    ///     (x: 110., y: 150.),
+    ///     (x: 160., y: 160.),
+    ///     (x: 125., y: 179.),
+    /// ];
+    ///
+    /// let intersection = line_string.relate(&poly);
+    /// assert_eq!(intersection.is_crosses(), true);
+    /// ```
+    ///
+    /// # Notes
+    /// - If any of the following do not hold, the function will return `false`:
+    ///     - The intersection of the interiors of the geometries must be non-empty
+    ///     - The intersection must have dimension less than the maximum dimension of the two input geometries (two polygons cannot cross)
+    ///     - The intersection of the two geometries must not equal either geometry (two points cannot cross)
+    /// - Matches one of `[T*T******] (a < b)`, `[T*****T**] (a > b)`, `[0********] (dimensions == 1)`
+    /// - This predicate is **symmetric and irreflexive**
+    pub fn is_crosses(&self) -> bool {
+        let dims_a = self.0[CoordPos::Inside][CoordPos::Inside]
+            .max(self.0[CoordPos::Inside][CoordPos::OnBoundary])
+            .max(self.0[CoordPos::Inside][CoordPos::Outside]);
+
+        let dims_b = self.0[CoordPos::Inside][CoordPos::Inside]
+            .max(self.0[CoordPos::OnBoundary][CoordPos::Inside])
+            .max(self.0[CoordPos::Outside][CoordPos::Inside]);
+        match (dims_a, dims_b) {
+            // a < b
+            (Dimensions::ZeroDimensional, Dimensions::OneDimensional)
```

Comment:
Alternatively, you can use an if rather than spelling out all the cases:
```
_ if dims_a < dims_b => {
                // [T*T******]
                self.0[CoordPos::Inside][CoordPos::Inside] != Dimensions::Empty
                    && self.0[CoordPos::Inside][CoordPos::Outside] != Dimensions::Empty
}
```

But what you have also looks correct.

Reply:
Ahh, I suspected there there was  a way to use `_` and an `if` statement to bundle these together.